octobot_trading/exchanges/util/exchange_util.py

```diff
@@ -66,12 +70,66 @@ def get_exchange_class_from_name(exchange_parent_class, exchange_name, tentacles
             # As we are searching for an exchange_type specific subclass
             # We should ignore classes that raises NotImplementedError
             pass
+    auto_filled_exchanges = _get_auto_filled_exchanges(tentacles_setup_config)
+    if exchange_name in auto_filled_exchanges:
+        return auto_filled_exchanges[exchange_name][0]
     return None
 
 
-def _is_exchange_candidate_matching(exchange_candidate, exchange_name, tentacles_setup_config, enable_default=False):
+def _get_auto_filled_exchanges(tentacles_setup_config):
+    auto_filled_exchanges = {}
+    for exchange_candidate in _get_auto_filled_exchanges_tentacles():
+        if tentacles_setup_config is None:
+            # tentacles_setup_config is required for auto-filled exchanges
+            continue
+        config = api.get_tentacle_config(
+            tentacles_setup_config, exchange_candidate
+        )
+        for exchange_name in exchange_candidate.supported_autofill_exchanges(config):
+            auto_filled_exchanges[exchange_name] = (exchange_candidate, config)
+    return auto_filled_exchanges
+
+
+def get_auto_filled_exchange_names(tentacles_setup_config):
+    return list(_get_auto_filled_exchanges(tentacles_setup_config))
+
+
+def _get_auto_filled_exchanges_tentacles():
+    return [
+        exchange_candidate
+        for exchange_candidate in tentacles_management.get_all_classes_from_parent(exchanges_types.RestExchange)
+        if exchange_candidate.HAS_FETCHED_DETAILS
+    ]
+
+
+async def get_exchange_details(
+    exchange_name, is_autofilled, tentacles_setup_config, aiohttp_session
+) -> exchange_details.ExchangeDetails:
+    if is_autofilled:
+        auto_filled_exchanges = _get_auto_filled_exchanges(tentacles_setup_config)
+        if exchange_name in auto_filled_exchanges:
+            return await auto_filled_exchanges[exchange_name][0].get_autofilled_exchange_details(
+                aiohttp_session, auto_filled_exchanges[exchange_name][1], exchange_name
+            )
+    try:
+        exchange = getattr(ccxt, exchange_name)()
+        return exchange_details.ExchangeDetails(
+            exchange.id,
+            exchange.name,
+            exchange.urls["www"],
+            exchange.urls["api"],
+            exchange.urls["logo"],
```

Comment:
Shouldn't we create an enum for these strings?

Reply:
right